### DISCUSSION TITLE
Nesting endpoints within shared parent

### DISCUSSION THREAD

**Discussion by kris-kuiper** [rebing/graphql-laravel#923] [Q&A]
Hi There, I'm trying to nest endpoints within a shared parent:
```graphql
{
  product {
    view(id: "R7YL5GJC4S4") {
      id
      title
      description
    }
  }
}
```

But the view type is never executed. So result is always:

```json
{
  "data": {
    "product": {
      "view": null
    }
  }
}
```

Code:
```php
class ProductQuery extends Query
{
    protected $attributes = [
        'name' => 'Product',
    ];

    public function type(): Type
    {
        return GraphQL::type('Product');
    }

    public function resolve($root, $args)
    {
        return GraphQL::type('View');
    }
}
```

Tried to do a couple of other things as well in the resolve method:

```php
public function resolve($root, $args)
{
    return [];
}
```

```php
public function resolve($root, $args)
{
    return function() {
        return [];
    };
}
```

Any idea's how to get this structure?

**Comment by mfn** [comment]
```
    public function resolve($root, $args)
    {
        return GraphQL::type('View');
    }
```
The resolver is expected to return the data (directly, closure, `\GraphQL\Deferred`), not a type definition/reference.

Please show the definition of all the types involved, currently I only see the query.

In the end, the `Product` type is expected to have a `view` field with your `View` type and it's respective fields (id, title, description).

The value return in `ProductQuery::resolve` is expected to match this shape: be it simply nested arrays or objects (the default field resolver tries to access the properties on objects.
